### PR TITLE
Replace depreacated react-router-redux by connected-react-router

### PR DESCRIPTION
**Pull request by Bnjis**
Hi i see the plugin react-router-redux was deprecated ([https://github.com/reactjs/react-router-redux](url)) 
I just replace it by connected-react-router. 

I give you the code if you ll need. 

Have a nice day ;)

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -4,14 +4,17 @@ import './assets/medium.css';
 import { Provider } from 'react-redux';
 
 import { Switch, Route } from 'react-router-dom';
-import { ConnectedRouter } from 'react-router-redux';
+//import { ConnectedRouter } from 'react-router-redux'; ==> OLD
```
Comment (dgr8akki) [inline src/index.js]: No need of comments here. Just remove it.

File: src/index.js [hunk 2]
```diff
@@ -4,14 +4,17 @@ import './assets/medium.css';
...
+import { ConnectedRouter } from "connected-react-router";
```
Comment (dgr8akki) [inline src/index.js]:
```suggestion
import { ConnectedRouter } from 'connected-react-router';
```

File: src/redux/reducer.js [hunk 3]
```diff
@@ -2,11 +2,13 @@ import { combineReducers } from 'redux';
 import articles from './reducers/articles';
 import authUser from './reducers/authUser';
 import common from './reducers/common';
-import { routerReducer } from 'react-router-redux';
+//import { routerReducer } from 'react-router-redux'; ==> OLD
```
Comment (dgr8akki) [inline src/redux/reducer.js]: No need of comments here. Just remove it.

File: src/redux/reducer.js [hunk 4]
```diff
@@ -2,11 +2,13 @@ import { combineReducers } from 'redux';
...
+import { connectRouter } from "connected-react-router";
```
Comment (dgr8akki) [inline src/redux/reducer.js]:
```suggestion
import { connectRouter } from 'connected-react-router';
```

File: src/redux/reducer.js [hunk 5]
```diff
@@ -2,11 +2,13 @@ import { combineReducers } from 'redux';
...
-  common,
-  router: routerReducer
-});
+export default history =>
+  combineReducers({
+    router: connectRouter(history),
+    articles,
+    authUser,
+    common
+  });
```
Comment (dgr8akki) [inline src/redux/reducer.js]: Add an extra empty line in the end of file.

File: src/redux/store.js [hunk 6]
```diff
@@ -1,15 +1,24 @@
-import { applyMiddleware, createStore } from 'redux';
+import { applyMiddleware, compose, createStore } from "redux";
 //import { createLogger } from 'redux-logger'
-import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
+// import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
```
Comment (dgr8akki) [inline src/redux/store.js]: No need of comments here. Just remove it.

File: src/redux/store.js [hunk 7]
```diff
@@ -1,15 +1,24 @@
-import { applyMiddleware, createStore } from 'redux';
+import { applyMiddleware, compose, createStore } from "redux";
```
Comment (dgr8akki) [inline src/redux/store.js]:
```suggestion
import { applyMiddleware, compose, createStore } from 'redux';
```

File: src/redux/store.js [hunk 8]
```diff
@@ -1,15 +1,24 @@
...
 //import { createLogger } from 'redux-logger'
-import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
+// import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
 import reducer from './reducer';
-import thunk from 'redux-thunk'
+import thunk from 'redux-thunk';
+import { routerMiddleware } from "connected-react-router";
 
-import createHistory from 'history/createBrowserHistory';
+import { createBrowserHistory } from "history";
```
Comment (dgr8akki) [inline src/redux/store.js]:
```suggestion
import { createBrowserHistory } from 'history';
```

File: src/redux/store.js [hunk 9]
```diff
@@ -1,15 +1,24 @@
...
+export default function configureStore(preloadedState) {
+  const store = createStore(
+    reducer(history),
+    preloadedState,
+    compose(applyMiddleware(routerMiddleware(history), thunk))
+  ); // root reducer with router state // for dispatching history actions
+  // ... other middlewares ...
+
+  return store;
+}
```
Comment (dgr8akki) [inline src/redux/store.js]: End of file

File: src/redux/reducer.js (hunk 3 above)
Comment (Bnjis) [reply src/redux/reducer.js]: Hi yeah maybe :) i do this long time ago when i learn React / Redux ;) Thix for the review

File: src/redux/reducer.js (hunk 4 above)
Comment (Bnjis) [reply src/redux/reducer.js]: Thx for review :)

File: src/redux/store.js (hunk 9 above)
Comment (Bnjis) [reply src/redux/store.js]: Yeah i think ur code is better ;)

**Pull request merged**